spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -175,7 +184,37 @@ public Statistics estimateStatistics() {
   protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
     if (snapshot == null) {
-      return new Stats(0L, 0L);
+      return new Stats(0L, 0L, Collections.emptyMap());
+    }
+
+    boolean cboEnabled =
+        Boolean.parseBoolean(spark.conf().get(SQLConf.CBO_ENABLED().key(), "false"));
+    Map<NamedReference, ColumnStatistics> colStatsMap = null;
+    if (readConf.enableColumnStats() && cboEnabled) {
+      colStatsMap = Maps.newHashMap();
+      List<StatisticsFile> files = table.statisticsFiles();
+      if (!files.isEmpty()) {
+        List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
+
+        for (BlobMetadata blobMetadata : metadataList) {
+          int id = blobMetadata.fields().get(0);
```

Comment:
Should we check if there is more than one  field here(for example, ndv stats is collected for say field1 and field2) and not propagate the stats if so?

Reply:
I can't find detailed documentation for `BlobMetadata` in the spec. My understanding is that for the blob type `apache-datasketches-theta-v1`, `ndv` is required. I assume that `ndv` is set in the `properties` as shown in the following example. If my assumption is correct, it seems to me that we can only have one field in the `fields`, establishing a one-to-one relationship with the `ndv` in the properties.
```
new GenericBlobMetadata(
    APACHE_DATASKETCHES_THETA_V1,
    snapshotId,
    1,
    ImmutableList.of(1),
    ImmutableMap.of("ndv", "4"))
```

Reply:
cc @findepi

Reply:
Correct, the `apache-datasketches-theta-v1` should be calculated on one field.
And yes, there should be the `ndv` property set. The property may seem somewhat redundant within the Puffin file, but allow faster access to the information at SELECT-time. More importantly, the properties are propagated to the table metadata and so a query planner accesses the NDV information without opening the Puffin file at all.